src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double INCREASE_CAPACITY = 1.5;
+    private static final int ARRAYS_ZERO_BORDER = 0;
+    private static final int INCREASE_INDEX_FOR_COPY = 1;
+    private T[] elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        if (elementData.length == size) {
+            elementData = grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (elementData.length == size) {
+            elementData = grow();
+        }
+        size++;
+        if (index < size && index >= ARRAYS_ZERO_BORDER) {
+            System.arraycopy(elementData, index, elementData,
+                    index + INCREASE_INDEX_FOR_COPY,
+                    size - (index + INCREASE_INDEX_FOR_COPY));
+            elementData[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index is bigger than arrays size");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (elementData.length < size + list.size()) {
+            elementData = grow();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            elementData[size] = list.get(i);
+            size++;
+        }
 
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < size && index >= ARRAYS_ZERO_BORDER) {
```

Comment:
lets move this check to separate method `if (index < size && index >= ARRAYS_ZERO_BORDER)`